pkg/core/consensus/selection/step.go

```diff
@@ -95,6 +98,21 @@ func (p *Phase) generateCandidate(ctx context.Context, round consensus.RoundUpda
 		WithField("round", round.Round).
 		Debugln("sending score")
 
+	if config.Get().Kadcast.Enabled {
```

Comment:
I am not a fan of littering the code with configuration checks that should only matter at startup (once the networking method is established we don't ever look back). Maybe we should consider using an interface that gets injected by the server at startup to the consensus loop. Will open a new issue

Reply:
I had this same idea. I've held off on it for now since my focus shifted to devnet, but I was considering this as well.